controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -1852,6 +1852,53 @@ int bml_internal::register_nw_map_update_cb(BML_NW_MAP_QUERY_CB pCB)
     return (BML_RET_OK);
 }
 
+int bml_internal::device_oper_radios_query(BML_DEVICE_DATA *device_data)
+{
+    // Initialize the promise for receiving the response
+    beerocks::promise<bool> prmDeviceDataGet;
+    m_prmDeviceDataGet = &prmDeviceDataGet;
+    int iOpTimeout     = RESPONSE_TIMEOUT; // Default timeout
+
+    beerocks_message::sDeviceData DeviceData;
+    m_device_data         = &DeviceData;
+    m_device_data->al_mac = network_utils::mac_from_string(device_data->al_mac);
+
+    register_topology_discovery_query();
+    // device radio status can be retrieved from topology discovery response
+    trigger_topology_discovery(device_data->al_mac);
+
+    int iRet = BML_RET_OK;
+
+    if (!prmDeviceDataGet.wait_for(iOpTimeout)) {
+        LOG(WARNING) << "Timeout while waiting for topology query response...";
+        iRet = -BML_RET_TIMEOUT;
+    }
+
+    unregister_topology_discovery_query();
```

Comment:
I agree with Moran here. If its possible to avoid extra mechanism and just be notified on each topology response (doesn't really have to be sent from the BML, can be eny response)

Reply:
I'm not familiar of any other way to signal bml_cli from son_master_thread (internal controller message)

Also couple of points for your suggestion, 
 1. It sounds wrong to go through the bus to send an internal controller message.
2. topology response sent periodically, do we want to put more load on the CPU  especially when this API will not be used in operational state.
4. signaling bml_task is faster.
5. we don't have eAction for controller->controller VS message (not much of a problem to add one but there is a reason to it)

also I'm not familiar of sending cannot send ACTION_BML directly from son